### PR TITLE
Added onResultStats Prop

### PR DESCRIPTION
**Pull request by bietkul**

### CODE REVIEW

File: packages/web/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 <details>
     <summary>The following changes have been included in the <code>dev</code> branch and will be out in the next release. <b>Click to expand</b></summary>
+     - Added `onResultStats` for reading purposes and `renderResultStats` for rendering purposes - [#461](https://github.com/appbaseio/reactivesearch/pull/461)
```
Comment (divyanshu013) [inline packages/web/CHANGELOG.md]: We should reference the issue in changelog here.

File: packages/native/src/components/result/ReactiveList.js [hunk 2]
```diff
@@ -130,7 +130,10 @@ class ReactiveList extends Component {
 		if (this.props.stream !== nextProps.stream) {
 			this.props.setStreaming(nextProps.componentId, nextProps.stream);
 		}
-
+		// called when results are updated
+		if (this.props.onResultStats && nextProps.hits !== this.props.hits) {
+			this.props.onResultStats(nextProps.total, nextProps.time);
```
Comment (metagrover) [inline packages/native/src/components/result/ReactiveList.js]: We have changed the spec for this. Refer: https://github.com/appbaseio/reactivesearch/pull/603

Comment (metagrover) [reply packages/native/src/components/result/ReactiveList.js]: Please also send a PR to update relevant docs in `v3` reactive-manual 🐱

**Comment by metagrover** [comment]
@bietkul can you please update this? 🐱

**Pull request closed without merging**